login-workflow/src/new-architecture/components/RegistrationWorkflow/RegistrationWorkflow.tsx

```diff
@@ -57,22 +63,62 @@ export const RegistrationWorkflow: React.FC<React.PropsWithChildren<Registration
         }
     };
 
+    const onNext = (): Promise<void> => {
+        const { firstName, lastName } = screenData.AccountDetails;
+        if (actions && actions().completeRegistration)
+            return actions()
+                .completeRegistration(
+                    { firstName, lastName },
+                    screenData.VerifyCode.code,
+                    screenData.CreateAccount.emailAddress
+                )
+                .then(({ email, organizationName }) => {
+                    updateScreenData({
+                        screenId: 'RegistrationSuccessScreen',
+                        values: { email, organizationName },
+                    });
+                    setShowSuccessScreen(true);
+                })
+                .catch((_error) => {
+                    setError({
+                        cause: {
+                            title: (_error as AuthError).cause.title,
+                            errorMessage: (_error as AuthError).cause.errorMessage,
+                        },
+                    });
+                });
+    };
+
     return (
         <RegistrationWorkflowContextProvider
             currentScreen={currentScreen}
             totalScreens={totalScreens}
-            nextScreen={(data): void => {
-                handleScreenNavigation(data);
+            nextScreen={(data): Promise<void> => {
+                updateScreenData(data);
+                if (currentScreen === totalScreens - 1) return onNext();
                 setCurrentScreen((i) => i + 1);
             }}
             previousScreen={(data): void => {
-                handleScreenNavigation(data);
+                updateScreenData(data);
                 setCurrentScreen((i) => i - 1);
             }}
             screenData={screenData}
-            updateScreenData={handleScreenNavigation}
+            updateScreenData={updateScreenData}
         >
-            {screens[currentScreen]}
+            {showSuccessScreen ? (
+                <RegistrationSuccessScreen
+                    errorDisplayConfig={{
```

Comment:
Do we need this? In my head, if we're on the SuccessScreen, there's no possible situation where we would need to display an error.

Reply:
Makes sense.